### ISSUE TITLE
Tooltips on hover behaviour is excessive 

### ISSUE THREAD

**Issue by Moshyfawn** [zed-industries/zed#8340]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Sorry for the generic title. This problem consists of a few behaviours and I don't know how to describe it in a few words.

When the mouse is moved over a syntax with a tooltip, but not hovered, the tooltip is displayed. This makes it look like the wrong information is being displayed relative to the current cursor position.

Also, this behaviour is inconsistent, but happens often enough to make it difficult to find the code, especially if you're new to it.

There's also a behaviour where if you move the cursor over a syntax and out of the editor window, the tooltip is permanently displayed. You have to move the cursor over another syntax to remove it, or press the Escape key.

### Environment

Zed: v0.124.4 (Zed Preview)
OS: macOS 14.2.1
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

https://github.com/zed-industries/zed/assets/16290753/51c55662-f3c0-464c-ab5f-ce60f23b158f

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

_No response_

**Comment by traderdude123** [comment]
Please give option to disable this.

Its very annoying.

**Comment by widersky** [comment]
@traderdude123 option is nice to have, but they need to just fix this, because it's obviously an issue that's very distracting while working

**Comment by babakgh** [comment]
This is very very annoying. I literally might stop using Zed since so many things pops up by just moving a mouse around. I started using Zed to develop in Ruby and it was awesome, since I moved my go codes to zed then it is a disaster with all "helps" that I didn't ask for it

**Comment by Moshyfawn** [comment]
> This is very very annoying. I literally might stop using Zed since so many things pops up by just moving a mouse around. I started using Zed to develop in Ruby and it was awesome, since I moved my go codes to zed then it is a disaster with all "helps" that I didn't ask for it

You can temporarily disable popovers by adding this to your `settings.json`:

```json
{
  "hover_popover_enabled": false
}
```

**Comment by florinpatrascu** [comment]
Thank you for allowing us to enable or disable the inline documentation, on hover. However, I'd like to suggest the possibility of introducing a `hover_popover_delay` feature. This would allow us to set a delay, measured in milliseconds, before the inline documentation popover appears. It would be particularly useful for users who highly value the inline documentation, offering them enhanced control over its display timing and not just an absolute condition. Many thanks.

**Comment by quintilation** [comment]
+1 for hover_popover_delay
the feature is useful, it just occurs too quickly

**Comment by Samuelodan** [comment]
I have a related problem where the popover doesn't stay long enough for me to navigate my mouse cursor to it. It disappears as soon as I move the cursor. A little bit of delay before disappearing will be helpful.

**Comment by ankhzet** [comment]
Is this merged? Tooltip is still very buggy for me (typescript, zed 0.146.5).
TLDR: Typehint tooltip is shown on keyboard interactions (even if it is completely disabled in settings), despite supposedly being a `hover` tooltiip.

Tried even turning the tooltip off completely in settings, to no avail:  ` "hover_popover_enabled": false` (~/.config/zed/settings.json, no local config)

How it's reproduced:
In editor:
```ts
16            <ProcessorPorts 
17                ports=| <- cursor here
18            />

...

23  const useStyles = createUseStyles({
24      label: {},
25  });

```
Step1: Typing `{` opens tooltip that shows `{ ports } ...`. _Why?_ More confusingly, event if tooltip is disabled in config it is still shown, it is not shown on hover, but _is_ shown upon typing.

Step 2.a: Next, if you without doing anything else besides first click, now click f.e. the line `23` (cursor ends up at the end of line after `{` symbol), the tooltip is now moved to that location, briefly shows previous contents (_why_?), and then updates to show `styles: ...`, _Why_? I didn't hover anything.
Step 2.b: If at line `17`, after typing`{` (tooltip appears) you continue typing or move the cursor with arrow keys, the tooltip would follow the cursor, never disappearing or changing contents, it would obscure the text (it covers completely the line above when you press newline, for example.), and if you dismiss it with Esc, it will immediately come back on next similar syntactic token (`{`, `"` etc), flickering and obscuring the code.

If I disable everything even theoretically related:
```json
  "show_completions_on_input": false,
  "show_completion_documentation": false,
  "completion_documentation_secondary_query_debounce": 30000,
  "use_on_type_format": false,
```

then the tooltip still appears in step 1 (though, it took two tries this time), still jumps to new location with cursor (still showing old contents contents) (in case 2.a, it stays only for a brief moment and then disappears, in case 2.b it follows cursor without closing).

Currently the tooltip is bugged and makes using the editor very taxing.

Overall, this tooltip should not be shown upon typing anything, it is very distracting and intrusive.
It should not follow the cursor. It should not keep old contents upon changing location.

**Comment by joeltg** [comment]
I've made the leap from VSCode to Zed a couple weeks ago but I'm going to have to switch back if this tooltip issue doesn't get resolved soon. The specific behavior described in the comment above - where the tooltip constantly appears during *keyboard* cursor navigation and doesn't disappear unless explicitly dismissed with Escape - is driving me absolutely insane.

**Comment by babakgh** [comment]
> I've made the leap from VSCode to Zed a couple weeks ago but I'm going to have to switch back if this tooltip issue doesn't get resolved soon. The specific behavior described in the comment above - where the tooltip constantly appears during *keyboard* cursor navigation and doesn't disappear unless explicitly dismissed with Escape - is driving me absolutely insane.
> 
> 
> 
> Zed 0.150.4 MacOS

Turn it off

https://github.com/zed-industries/zed/issues/8340#issuecomment-1970199044

**Comment by ankhzet** [comment]
> Turn it off
> 
> [#8340 (comment)](https://github.com/zed-industries/zed/issues/8340#issuecomment-1970199044)

My comment explicitly says, that this behavior persists _despite the tooltip and ALL related behaviour being TURNED OFF in settings_.
